components/input/src/auro-input.js

```diff
@@ -0,0 +1,213 @@
+
+    /**
+     * @private
+     */
+    this.buttonTag = versioning.generateTag('auro-button', buttonVersion, AuroButton);
+  }
+
+  /**
+   * Function to determine if the input is meant to render an icon visualizing the input type.
+   * @private
+   * @returns {boolean} - Returns true if the input type is meant to render an icon.
+   */
+  hasTypeIcon() {
+    const typesWithIcons = [
+      'month-day-year',
+      'month-year',
+      'year-month-day',
+      'month-fullYear'
+    ];
+
+    if (this.icon || typesWithIcons.includes(this.type)) {
+      return true;
+    }
+
+    return false;
+  }
+
+  // function that renders the HTML and CSS into  the scope of the component
+  render() {
+    // is-disabled class - THIS IS ONLY HERE TO MAKE A TEST PASS AS FAR AS I CAN TELL
+    const labelClasses = {
+      'is-disabled': this.disabled,
+      'withIcon': this.hasTypeIcon(),
+      'withValue': this.value && this.value.length > 0
+    };
+
+    return html`
+      <div class="wrapper" part="wrapper">
+        <div class="main">
+          <div class="typeIcon">
+            ${this.type === 'credit-card' ? this.processCreditCard() : undefined}
+
+            <!-- The repeat() method is used below in order to force auro-icon to re-render when the name value is updated.
+               This should be cleaned up when auro-icon issue #31 is resolved. -->
+            ${this.inputIconName
+            ? repeat([this.inputIconName], (val) => val, (name) => html`
+              <${this.iconTag}
+                class="accentIcon"
+                category="payment"
+                name="${name}"
+                part="accentIcon"
+                ?disabled="${this.disabled}">
+              </${this.iconTag}>
+            `) : undefined
+            }
+
+            ${this.type === 'month-day-year' || this.type === 'month-year' || this.type === 'year-month-day' || this.type === 'month-fullYear'
+            ? html`
+              <${this.iconTag}
+                class="accentIcon"
+                category="interface"
+                name="calendar"
+                part="accentIcon"
+                ?disabled="${this.disabled}">
+              </${this.iconTag}>`
+            : undefined
+            }
+          </div>
+          <label for=${this.id} class="${classMap(labelClasses)}" part="label">
+            <slot name="label">
+              ${this.label}
+            </slot>
+            ${this.required ? '' : ` (${i18n(this.lang, 'optional')})`}
+          </label>
+          <input
+            @input="${this.handleInput}"
+            @focusin="${this.handleFocusin}"
+            @blur="${this.handleBlur}"
+            id="${this.id}"
+            name="${ifDefined(this.name)}"
+            type="${this.type === 'password' && this.showPassword ? 'text' : this.getInputType(this.type)}"
+            pattern="${ifDefined(this.definePattern())}"
+            maxlength="${ifDefined(this.maxLength ? this.maxLength : undefined)}"
```

Comment:
Instead of using `"${ifDefined(this.maxLength || undefined)}"`, you can use this syntax:

`?maxLength="this.maxLength"`

The `?` essentially acts as an `if` statement and applies the attribute to the component if it exists. You can do this anywhere you see `ifDefined`. It would be good to refactor this because we do not do that in any other component.

Reply:
Out of scope for this PR, but I have captured it here in this `auro-input` ticket: https://github.com/AlaskaAirlines/auro-input/issues/353